var/spack/repos/builtin/packages/elk/package.py

```diff
@@ -29,61 +36,66 @@ class Elk(Package):
     # Cannot be built in parallel
     parallel = False
 
-
     def configure(self, spec):
         # Dictionary of configuration options
         config = {
-            'MAKE':      'make',
-            'F90':       join_path(spack.build_env_path, 'f90'),
-            'F77':       join_path(spack.build_env_path, 'f77'),
-            'AR':        'ar',
-            'LIB_FFT':   'fftlib.a',
-            'SRC_MPI':   'mpi_stub.f90',
-            'SRC_OMP':   'omp_stub.f90',
-            'SRC_libxc': 'libxcifc_stub.f90',
-            'SRC_FFT':   'zfftifc.f90'
+            'MAKE': 'make',
+            'AR':   'ar'
         }
 
         # Compiler-specific flags
         flags = ''
-        if   self.compiler.name == 'intel':
-            flags = '-O3 -ip -unroll -no-prec-div -openmp'
+        if self.compiler.name == 'intel':
+            flags = '-O3 -ip -unroll -no-prec-div'
         elif self.compiler.name == 'gcc':
-            flags = '-O3 -ffast-math -funroll-loops -fopenmp'
+            flags = '-O3 -ffast-math -funroll-loops'
         elif self.compiler.name == 'pgi':
-            flags = '-O3 -mp -lpthread'
+            flags = '-O3 -lpthread'
         elif self.compiler.name == 'g95':
             flags = '-O3 -fno-second-underscore'
         elif self.compiler.name == 'nag':
             flags = '-O4 -kind=byte -dusty -dcfuns'
         elif self.compiler.name == 'xl':
-            flags = '-O3 -qsmp=omp'
+            flags = '-O3'
         config['F90_OPTS'] = flags
         config['F77_OPTS'] = flags
 
         # BLAS/LAPACK support
+        # Note: BLAS/LAPACK must be compiled with OpenMP support
+        # if the +openmp variant is chosen
         blas   = 'blas.a'
         lapack = 'lapack.a'
-        if '+blas'   in spec:
-            blas   = join_path(spec['blas'].prefix.lib,   'libblas.so')
+        if '+blas' in spec:
+            blas   = spec['blas'].blas_shared_lib
         if '+lapack' in spec:
-            lapack = join_path(spec['lapack'].prefix.lib, 'liblapack.so')
-        config['LIB_LPK'] = ' '.join([lapack, blas]) # lapack must come before blas
+            lapack = spec['lapack'].lapack_shared_lib
+        # lapack must come before blas
```

Comment:
What compiler "searches for symbols both ways"?  AFAIK nearly everything calls `ld` underneath.  I don't know of a linker that searches both ways, and doing so would be confusing for things like profiling libraries that need to be interposed at particular spots.  e.g., `PMPI` libraries like [mpip](http://mpip.sourceforge.net) need to come *before* MPI on the link line b/c they intercept MPI routines.  If the linker can resolve in any order, you can't be guaranteed your profiling library will even work.

There *are* weird things like [`--start-group` and `--end-group`](http://stackoverflow.com/questions/5651869/gcc-what-are-the-start-group-and-end-group-command-line-options).  But that is a mechanism to handle circular dependencies :scream:, and I think the linker searches repeatedly left-to-right to make this work, and not in arbitrary order.